File: product-service/src/main/java/com/ezbuy/productservice/client/config/WebClientFactoryConfig.java
package com.ezbuy.productservice.client.config;

import com.ezbuy.productservice.client.properties.*;
import com.reactify.client.WebClientFactory;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@RequiredArgsConstructor
public class WebClientFactoryConfig {
    private final AuthProperties authProperties;
    private final OrderProperties orderProperties;
    private final SettingProperties settingProperties;

    @Bean(name = "webClientFactory")
    public WebClientFactory webClientFactory() {
        return new WebClientFactory(List.of(authProperties, orderProperties, settingProperties));
    }
}

File: notification-service/src/main/kotlin/com/ezbuy/notificationservice/client/config/WebClientFactoryConfig.kt
package com.ezbuy.notificationservice.client.config

import com.ezbuy.notificationservice.client.properties.AuthProperties
import com.reactify.client.WebClientFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class WebClientFactoryConfig(private val authProperties: AuthProperties) {
    @Bean(name = ["webClientFactory"])
    fun webClientFactory(): WebClientFactory {
        return WebClientFactory(listOf(authProperties))
    }
}
